### ISSUE TITLE
setup.py tweaks for Mingw on Windows

### ISSUE THREAD

**Issue by vineetbansal** [flatironinstitute/finufft#47]
I had to make these small tweaks to setup.py to be be able to build the finufft extension on Windows x64. There are some minor changes needed in the makefile to be able to generate the .pyd successfully using `make lib`, but that's another discussion.

As for setup.py, the following is the minimal set of changes. The first block if for good-faith defaults for Windows, and the second to get around a well-known nuance of Mingw32. (see https://stackoverflow.com/questions/10660524/error-building-boost-1-49-0-with-gcc-4-7-0/12124708#12124708)

```
@@ -31,7 +31,12 @@ class get_pybind_include(object):
         return pybind11.get_include(self.user)

 # choose compile flags for finufftpy.cpp (links to finufft lib)...
-if sys.platform == "linux" or sys.platform == "linux2":
+if sys.platform == "win32":
+    libraries = ["lib-static/finufft","fftw3"]
+    extra_compile_args=['-fopenmp']
+    extra_link_args=[]
+
+elif sys.platform == "linux" or sys.platform == "linux2":
     # changed from fftw3_threads, since a bit faster, 9/24/18:
     libraries = ["lib-static/finufft","fftw3","fftw3_omp","gomp"]
     extra_compile_args=['-fopenmp']
@@ -99,6 +104,7 @@ class BuildExt(build_ext):
     c_opts = {
         'msvc': ['/EHsc'],
         'unix': [],
+        'mingw32': ['-D_hypot=hypot']
     }

     # Mac OSX w/ GCC (not clang) you may need to comment out the next two lines:
```

I've posted a built win_x64 binary wheel for finufft in case it's useful for your testing:
https://file.io/EcRID5

**Comment by ahbarnett** [comment]
Dear Vineet,
Thanks - I pushed your patch.
But the file.io link was broken.
Now I'm having new python3 troubles with my miniconda - it can't import finufftpy_cpp.
python2 is still fine though.
(I am not a python expert).
Best, Alex

**Comment by vineetbansal** [comment]
Looks like the file.io link expired. I've moved the .whl file to a more permanent location:
https://www.dropbox.com/s/csbs4aarfwdntap/finufftpy-1.1.1-cp36-cp36m-win_amd64.whl?dl=0

Are your python3 issues related to the patch? If so my apologies and I'm happy to fix these - I built the binary wheel on Python 3.6 on Windows as well as Python 3.6 on Linux using the same patch and things worked fine.
